### PR TITLE
Change binops for-each kernel to thrust::for_each_n

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Replaces the custom `for_each_kernel` in `binary_ops.cuh` with `thrust::for_each_n`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by davidwendt** [comment]
Binary-ops benchmark runs comparison for RTX A6000
[binops-benchmarks.txt](https://github.com/user-attachments/files/17872980/binops-benchmarks.txt)

**Comment by davidwendt** [comment]
New results using `exec_policy_nosync` show much better results overall.
[binops-benchmarks2.txt](https://github.com/user-attachments/files/17874104/binops-benchmarks2.txt)

File: cpp/src/binaryop/compiled/binary_ops.cuh [hunk 1]
```diff
@@ -244,24 +244,6 @@ struct binary_op_double_device_dispatcher {
...
-  auto const stride = cudf::detail::grid_1d::grid_stride();
-
-#pragma unroll
-  for (auto i = start; i < size; i += stride) {
-    f(i);
-  }
-}
-
 /**
  * @brief Launches Simplified for_each kernel with maximum occupancy grid dimensions.
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/src/binaryop/compiled/binary_ops.cuh:248]: Yep, I'll remove it.

**Review by ttnghia — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Comment by karthikeyann** [comment]
Interesting!
https://github.com/NVIDIA/cccl/pull/1302 PR introduced `cub::DeviceFor::ForEachN` and further improvements. It's a good idea to consider replacing all other custom for_each like kernels with `thrust::for_each`

**Pull request merged**